pkg/metrics/metrics.go

```diff
@@ -224,11 +224,20 @@ func (c *Config) Configure(ctx context.Context) error {
 	}
 	// In the case of an operator restart, a previous Endpoint object will be deleted and a new one will
 	// be created to ensure we have a correct spec.
-	err = c.CoreV1().Endpoints(c.namespace).Delete(ctx, WindowsMetricsResource, metav1.DeleteOptions{})
-	if err != nil && !apierrors.IsNotFound(err) {
-		return errors.Wrap(err, "error deleting existing metrics Endpoint")
+	var subsets []v1.EndpointSubset
+	existingEndpoint, err := c.CoreV1().Endpoints(c.namespace).Get(ctx, WindowsMetricsResource, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return errors.Wrap(err, "error retrieving existing metrics endpoint")
+		}
+	} else {
+		subsets = existingEndpoint.Subsets
```

Comment:
Can `existingEndpoint` be nil such that we would try to access `.Subsets` of a nil pointer obj? Looks like no based on my understanding of the client API, but just want another set of eyes to make sure.

Reply:
I dont see how, if Get() is returning no error, existingEndpoint shouldnt be nil